clang/test/CodeGenHLSL/loops/unroll.hlsl

```diff
@@ -0,0 +1,99 @@
+// RUN: %clang_cc1 -std=hlsl2021 -finclude-default-header -x hlsl -triple \
+// RUN: dxil-pc-shadermodel6.3-library %s -emit-llvm -o - | FileCheck %s
+
+/*** for ***/
+void for_count()
+{
+// CHECK-LABEL: for_count
```

Comment:
We are not actually checking that these loops get unrolled. I think that is mostly ok, but I'm wondering if there are other mismatches between the expected hlsl behavior and the llvm loop unroller.

Are we guaranteed that the loop will be unrolled if we have the loop count metadata added? Will llvm raise an error if the loop cannot be unrolled when it has the unroll directive?

Reply:
`Are we guaranteed`
No these are typically just hints, also  these tests aren't running without any optimizations, i'll make that more explicit by adding `-disable-llvm-passes` back.